sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathAccessControlEntry.cs

```diff
@@ -0,0 +1,233 @@
+            {
+                throw new ArgumentException("AccessControlType must be User or Group if entityId is specified.");
+            }
+
+            DefaultScope = defaultScope;
+            AccessControlType = accessControlType;
+            EntityId = entityId;
+            Permissions = permissions;
+        }
+
+        /// <summary>
+        /// Override of ToString().
+        /// </summary>
+        /// <returns></returns>
+        public override string ToString()
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+
+            if (DefaultScope)
+            {
+                stringBuilder.Append("default:");
+            }
+            stringBuilder.Append(AccessControlType.ToString().ToLowerInvariant());
+            stringBuilder.Append(":");
+            stringBuilder.Append(EntityId ?? "");
+            stringBuilder.Append(":");
+            stringBuilder.Append(Permissions.ToSymbolicString());
+
+            return stringBuilder.ToString();
+        }
+
+        /// <summary>
+        /// Parses the provided string into a <see cref="PathAccessControlEntry"/>
+        /// </summary>
+        /// <param name="aclString">The string representation of the access control list.</param>
+        /// <returns>A <see cref="PathAccessControlEntry"/>.</returns>
+        public static PathAccessControlEntry Parse(string aclString)
+        {
+            if (aclString == null)
+            {
+                return null;
+            }
+
+            PathAccessControlEntry entry = new PathAccessControlEntry();
+            string[] parts = aclString.Split(':');
+            int indexOffset = 0;
+
+            if (parts.Length < 3 || parts.Length > 4)
+            {
+                throw new ArgumentException("aclString should have 3 or 4 parts delimited by colons");
+            }
+
+            if (parts.Length == 4)
+            {
+                if (!parts[0].Equals("default", StringComparison.OrdinalIgnoreCase))
+                {
+                    throw new ArgumentException("If aclString is 4 parts, the first must be \"default\"");
+                }
+                entry.DefaultScope = true;
+                indexOffset = 1;
+            }
+            entry.AccessControlType = ParseAccesControlType(parts[indexOffset]);
+
+            if (!string.IsNullOrEmpty(parts[1 + indexOffset]))
+            {
+                entry.EntityId = parts[1 + indexOffset];
+            }
+
+            entry.Permissions = RolePermissionsExtensions.ParseSymbolic(parts[2 + indexOffset], false);
+            return entry;
+        }
+
+        /// <summary>
+        /// Converts the Access Control List to a <see cref="string"/>.
+        /// </summary>
+        /// <param name="accessControlList">The Access Control List to serialize</param>
+        /// <returns>string.</returns>
+        public static string SerializeList(IList<PathAccessControlEntry> accessControlList)
```

Comment:
We shouldn't call this `SerializeList` - it doesn't really match our ToString/Parse methods everywhere else.  Should we create a static `PathAccessControl` for this and `DeserializeList`?

Reply:
We could, or we could rename to follow the ToString/Parse convention.

I think I your suggestion better, to move to a new static PathAccessControl class.

Reply:
I think we should match the ToString/Parse names so this feels consistent.

Reply:
fixed.